### PR TITLE
prefix with mz{resource_id} from Materialize CR status

### PR DESCRIPTION
**Pull request by alex-hunt-materialize**
Prefix kubernetes object names with `mz{resource_id}-` based on a randomly generated string stored in the Materialize CR status.

### Motivation

Eventual support for multiple Materialize CRs in a single namespace. We need to avoid name collisions, and we need something short and consistently starting with a letter to avoid DNS-1035 issues.

Originating slack discussion: https://materializeinc.slack.com/archives/CL68GT3AT/p1730717511313559

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered. ([trigger-ci for additional test/nightly runs](https://trigger-ci.dev.materialize.com/))
- [x] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [x] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [x] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [x] If this PR includes major [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note), I have pinged the relevant PM to schedule a changelog post.

### CODE REVIEW

File: src/orchestrator-kubernetes/src/lib.rs [hunk 1]
```diff
@@ -1690,7 +1698,7 @@ impl OrchestratorWorker {
 
     async fn list_services(&self, namespace: &str) -> Result<Vec<String>, K8sError> {
         let stateful_sets = self.stateful_set_api.list(&Default::default()).await?;
-        let name_prefix = format!("{namespace}-");
+        let name_prefix = format!("{}{namespace}-", self.name_prefix);
```
Comment (doy-materialize) [inline src/orchestrator-kubernetes/src/lib.rs:1701]: i forgot that we use `namespace` to mean something entirely different in orchestrator-kubernetes, this was pretty confusing for a sec

**Review by doy-materialize — APPROVED** [review]
to clarify my understanding, the reason this works is that we have to deploy a new environmentd version to pick up the changes to the command line flags, and old pods/services/etc are cleaned up by owner references to the old environmentd pod, not by trying to look up any existing resources by name

File: src/cloud-resources/src/crd/materialize.rs [hunk 2]
```diff
@@ -27,6 +28,8 @@ pub const LAST_KNOWN_ACTIVE_GENERATION_ANNOTATION: &str =
     "materialize.cloud/last-known-active-generation";
 
 pub mod v1alpha1 {
+    use rand::distributions::Uniform;
```
[the remark this answers is not in the document]
Comment (alex-hunt-materialize) [reply src/cloud-resources/src/crd/materialize.rs:31]: Ah, yes. This was an auto-import thing I forgot to move.

File: src/orchestrator-kubernetes/src/lib.rs (hunk 1 above)
Comment (alex-hunt-materialize) [reply src/orchestrator-kubernetes/src/lib.rs:1701]: Yeah, it was a really poor choice of terminology.

**Comment by alex-hunt-materialize** [comment]
>to clarify my understanding, the reason this works is that we have to deploy a new environmentd version to pick up the changes to the command line flags, and old pods/services/etc are cleaned up by owner references to the old environmentd pod, not by trying to look up any existing resources by name

I think that is mostly accurate, but there is at least one place we lookup by name, and I have updated that in this PR.

File: src/cloud-resources/src/crd/materialize.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (alex-hunt-materialize) [reply src/cloud-resources/src/crd/materialize.rs:31]: Yes, but I forgot to push before merging so here's a separate PR https://github.com/MaterializeInc/materialize/pull/30350

**Pull request merged**